### PR TITLE
Spring Cleaning: Part 1

### PR DESCRIPTION
**Pull request by ghost**
Cleaning of the first 5 classes.
- ht_bitfield.h
- ht_debug.h
- ht_file.h
- ht_file_exception.h
- ht_file_interface.h

### CODE REVIEW

File: include/ht_file_interface.h [hunk 1]
```diff
@@ -14,49 +14,200 @@
...
-
+        /**
+        \interface IFile
+        \ingroup HatchitCore
+        \brief An interface to describe system files.
+        
+        Interface describes the common functionality of files.
+        
+        **/
         class HT_API IFile : public INonCopy
```
Comment (ghost) [inline include/ht_file_interface.h:35]: I believe we didn't want interfaces to inherit from each other, so should we move this inheritance up to the File class?

Comment (Honeybunch) [inline include/ht_file_interface.h:35]: It's fine for now. If we decide we want that it can be in another PR
